src/datasources/entity-ds/queries/queryBuilder.ts

```diff
@@ -161,13 +161,16 @@ export const getTemplateVariable = (templateSrv: any, name: string) => {
     return undefined
 }
 
+const isValidClauseRestriction = (clause) => {
+    return clause?.restriction?.attribute && clause?.restriction?.comparator && clause?.restriction?.value
+}
+
 // Given a query filter, build a cloned one which does template variable substitution, etc.
 // Clone Filter to make substitution possible
 // (otherwise substitution would happen in original query,
 // and overwriting the $<variable> or [[variable]] in restrictions which may not be the intention)
 export const buildQueryFilter = (filter: API.Filter, request: EntityQueryRequest<EntityQuery>, templateSrv: TemplateSrv): API.Filter => {
-
-    filter.clauses = filter.clauses.filter((clause, idx) => clause.restriction?.attribute && clause.restriction?.comparator && clause.restriction?.value)
+    filter.clauses = filter.clauses.filter(clause => isValidClauseRestriction(clause))
```

Comment:
I know I'm a little late to the approval game here, but I just wanted to call out this small refactor for making me happy.